content/4_Gold/DSU.mdx

```diff
@@ -269,32 +301,33 @@ public class Main {
 			}
 			return true;
 		}
+
+		// returns whether two nodes are in the same connected component
+		public boolean connected(int x, int y) {
+			return find(x) == find(y);
+		}
 	}
 
-	public static void main(String[] args) throws IOException {
-		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
-		PrintWriter out = new PrintWriter(System.out);
-		StringTokenizer st = new StringTokenizer(br.readLine());
-		int size = Integer.parseInt(st.nextToken());
-		int queryNum = Integer.parseInt(st.nextToken());
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
```

Comment:
why scanner 👀 i thought this was discouraged?

Reply:
Should I switch to Kattio? I used scanner here cause it was fast enough (it's also the simplest).

Reply:
i mean, i guess it's fine if it's fast enough?? but idk

Reply:
okay I can switch to kattio